FlowCrypt/Functionality/Services/GlobalRouter.swift

```diff
@@ -36,75 +36,76 @@ final class GlobalRouter {
         return delegate.window
     }
 
-    private let userAccountService: UserAccountServiceType
-    private let googleService: GoogleUserService
-
     private lazy var logger = Logger.nested(in: Self.self, with: .userAppStart)
 
-    init(
-        userAccountService: UserAccountServiceType = UserAccountService(),
-        googleService: GoogleUserService = GoogleUserService()
-    ) {
-        self.userAccountService = userAccountService
-        self.googleService = googleService
-    }
 }
 
 // MARK: - Proceed
 extension GlobalRouter: GlobalRouterType {
+
     /// proceed to flow (signing/setup/app) depends on user status (isLoggedIn/isSetupFinished)
     func proceed() {
-        validateEncryptedStorage {
-            userAccountService.cleanupSessions()
-            proceed(with: nil)
+        do {
+            let appContext = try AppContext.setUpAppContext(globalRouter: self)
+            do {
+                try appContext.encryptedStorage.validate()
+                proceed(with: appContext)
+            } catch {
+                renderInvalidStorageView(error: error, encryptedStorage: nil)
+            }
+        } catch {
+            renderInvalidStorageView(error: error, encryptedStorage: nil)
         }
     }
```

Comment:
Main part of the PR. Set up context first, then pass it around. Don't re-setup everything in many places.